src/main/java/com/fulcrumgenomics/util/DemuxFastqs.java

```diff
@@ -677,14 +697,24 @@ void convertQuality(final byte[] quals, final FastqQualityFormat version) {
             return sampleBarcodeQualities;
         }
 
-        private String getMolecularBarcode(final List<FastqRecord>  records) {
-            final byte[][] molecularBarcode = new byte[records.size()][];
+        private byte[][] getMolecularBarcodes(final List<FastqRecord> records) {
+            final byte[][] molecularBarcodes = new byte[records.size()][];
```

Comment:
You could still return the byte[][] at the end, I'm suggesting that the temporary thing be a List, then it can be dynamic in size and at the end you can call `list.toArray(new byte[list.size]);`